### PR TITLE
 Replace SentinelAPIError with a more specific exception in all foreseeable cases

### PR DESCRIPTION
**Pull request by valgur**
WIP, extends #275 

Moved exceptions to a separate `exceptions.py` file and created a bunch of new ones to cover all foreseeable error cases. All of the added exceptions inherit from `SentinelAPIError`, for greater flexibility and backwards compatibility.

The list of exceptions after these changes is
* `SentinelAPIError` - the parent, catch-all exception. Only used when no other more specific exception can be applied.
* `SentinelAPILTAError` - error raised when retrieving a product from the Long Term Archive.
* `ServerError` - error raised when the server responded in an unexpected manner, typically due to undergoing maintenance.
* `UnauthorizedError` - error raised when attempting to retrieve a product with incorrect credentials.
* `QuerySyntaxError` - error raised when the query string could not be parsed on the server side.
* `QueryLengthError` - error raised when the query string length was excessively long.
* `InvalidKeyError` - error raised when product with given key was not found on the server.
* `InvalidChecksumError` - MD5 checksum of a local file does not match the one from the server.

TODO:
- [x] update docs
- [x] add changelog entry

### CODE REVIEW

File: sentinelsat/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        self.msg = msg
+        self.response = response
+
+    def __str__(self):
+        return 'HTTP status {0} {1}: {2}'.format(
+            self.response.status_code, self.response.reason,
+            ('\n' if '\n' in self.msg else '') + self.msg)
+
+
+class SentinelAPILTAError(SentinelAPIError):
```
Comment (valgur) [inline sentinelsat/exceptions.py]: How about renaming `SentinelAPILTAError` to `LongTermArchiveError`?

File: sentinelsat/__init__.py [hunk 2]
```diff
@@ -2,12 +2,9 @@
 
 # Import for backwards-compatibility
 from . import sentinel
-
+from .exceptions import *
```
Comment (j08lue) [inline sentinelsat/__init__.py:5]: Perhaps be explicit about the imports here, even though we are just in `__init__.py`

File: sentinelsat/scripts/cli.py [hunk 3]
```diff
@@ -6,7 +6,8 @@
 import requests.utils
 
 from sentinelsat import __version__ as sentinelsat_version
-from sentinelsat.sentinel import SentinelAPI, SentinelAPIError, geojson_to_wkt, read_geojson
+from sentinelsat.exceptions import *
```
Comment (j08lue) [inline sentinelsat/scripts/cli.py:9]: No wildcard imports, please.

File: sentinelsat/sentinel.py [hunk 4]
```diff
@@ -20,10 +20,11 @@
 import geomet.wkt
 import html2text
 import requests
-from six import string_types
+from six import string_types, raise_from
 from six.moves.urllib.parse import urljoin, quote_plus
 from tqdm import tqdm
 
+from sentinelsat.exceptions import *
```
Comment (j08lue) [inline sentinelsat/sentinel.py:27]: Again, no wildcard imports.

File: tests/test_download.py [hunk 5]
```diff
@@ -11,7 +11,8 @@
 import pytest
 import requests_mock
 
-from sentinelsat import SentinelAPI, SentinelAPILTAError, InvalidChecksumError, SentinelAPIError
+from sentinelsat import SentinelAPI
+from sentinelsat.exceptions import *
```
Comment (j08lue) [inline tests/test_download.py:15]: No wildcard imports.

File: tests/test_misc.py [hunk 6]
```diff
@@ -5,7 +5,8 @@
 import requests
 import requests_mock
 
-from sentinelsat import SentinelAPI, SentinelAPIError
+from sentinelsat import SentinelAPI
+from sentinelsat.exceptions import *
```
Comment (j08lue) [inline tests/test_misc.py:9]: Same

File: tests/test_opensearch.py [hunk 7]
```diff
@@ -1,11 +1,15 @@
 """
 Tests for functionality related to the OpenSearch API of SciHub (https://scihub.copernicus.eu/apihub/search?...)
 """
+from contextlib import contextmanager
 from datetime import datetime, date, timedelta
 
 import pytest
+import requests_mock
+from requests.exceptions import *
```
Comment (j08lue) [inline tests/test_opensearch.py:9]: Ditto

File: tests/test_opensearch.py [hunk 8]
```diff
@@ -1,11 +1,15 @@
...
 
-from sentinelsat import format_query_date, SentinelAPIError, SentinelAPI
+from sentinelsat import format_query_date, SentinelAPI
+from sentinelsat.exceptions import *
```
Comment (j08lue) [inline tests/test_opensearch.py:12]: Woop

**Comment by Fernerkundung** [comment]
I won't find time to review this and don't want to block this PR. I think it's fine if there is 1 reviewer that did a pass over the changes.

**Comment by valgur** [comment]
As I was already reorganizing the docs in this PR, I also included a format change for the CLI Sphinx docs in the last commits.
I think it's easier to read, can be more detailed and can be linked to with cross-references.
Screenshot from 2020-06-30 21-32-58

**Comment by j08lue** [comment]
Ah, sorry, was I too fast merging this? It still said "WIP" in the description...

**Comment by valgur** [comment]
Ah, no, I simply to forgot the PR description. Everything should be complete and covered, as far as I'm aware.

**Pull request merged**